pipeline dsview

hana-alice opened the pull request:
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

<!-- Note: Makes sure these boxes are checked before submitting your PR - thank you!
- [ ] Your pull request title is using English, it's precise and appropriate.
- [ ] If your pull request has gone "stale", you should **rebase** your work on top of the latest version of the upstream branch.
- [ ] If your commit history is full of small, unimportant commits (such as "fix pep8" or "update tests"), **squash** your commits down to a few, or one, discreet changesets before submitting a pull request.
- [ ] Document new code with comments in source code based on API docs
- [ ] Make sure any runtime log information in `log` , `error` or `new Error('')` has been moved into `EngineErrorMap.md` with an ID, and use `logID(id)` or `new Error(getError(id))` instead.
- To official teams:
  - [ ] Check that your PR is following our [guides](https://github.com/cocos/3d-tasks/blob/master/workflows/readme.md)
-->

SantyWang committed:
Merge pull request #14818 from cocos/develop

Sync develop to marionette

shrinktofit committed:
Introduce pose graph, pose node and pose state (#14779)

* Introduce pose graph, pose node and pose state

* Optimize for-of

* Rename

* Doc

* Rename pose allocator -> pose stack allocator

shrinktofit committed:
Merge remote-tracking branch 'upstream/develop' into marionette-next

SantyWang committed:
Merge pull request #14832 from shrinktofit/marionette-next

Sync develop to marionette-next

shrinktofit committed:
Animation graph: move code (#14819)

shrinktofit committed:
Support more binary condition binding source (#14857)

* Binary condition

* Update

* Fix test & i18n

shrinktofit committed:
Fix misc errors in marionette editor modules (#14907)

shrinktofit committed:
Animation graph custom events listening API (#14911)

SantyWang committed:
Merge pull request #14927 from cocos/develop

Sync develop into marionette-next

shrinktofit committed:
Remove duplicated test (#14975)

shrinktofit committed:
Animation graph: state machine adjustment (#14990)

* Animation graph: state machine adjustment

* Remove useless code; export factory interfaces

* Add test

* Update

shrinktofit committed:
Optimize TCBinding, implement TCStateWeightBinding, add TCStateMotionTimeBinding (#15009)

shrinktofit committed:
Optimize warn message if state machine matched too many transitions in a frame (#15016)

shrinktofit committed:
Complementarily implement pose graph (#14952)

* Implement pose graph

* Add test for query output node's input/outputs

* Fix isWellFormedInputKey()

* Tweak

* Use `Map` instead of `shellTag`

* Rename XNode -> PureValueNode

* Rename node input manager and node input options

shrinktofit committed:
Animation graph: experimentally support more types of variable (#14815)

* Animation graph: add experimental more types of variable

* Tweak

minggo committed:
Merge pull request #15067 from cocos/develop

Merge develop into marionette-next

shrinktofit committed:
Animation graph: optimize some evaluation structures (#15049)

* Optimize some evaluation structure

* Update cocos/animation/marionette/state-machine/state-machine-eval.ts

shrinktofit committed:
Merge remote-tracking branch 'upstream/develop' into marionette-next

minggo committed:
Merge pull request #15101 from shrinktofit/marionette-next

Merge remote-tracking branch 'upstream/develop' into marionette-next

shrinktofit committed:
Animation: lint and tweak (#15104)

cocos-robot committed:
[ci skip][AUTO]: Automated code generating update: 464342eb99a9f9d9d6df3759f2c9622a88b8152f  (#15101) (#15102)

Co-authored-by: cocos-robot <cocos-robot@cocos.com>

pandamicro committed:
Merge pull request #15113 from cocos/develop

Merge remote-tracking branch 'upstream/develop' into marionette-next

shrinktofit committed:
Merge remote-tracking branch 'upstream/develop' into marionette-next

VisualSJ committed:
Merge pull request #15127 from shrinktofit/marionette-next

Merge remote-tracking branch 'upstream/develop' into marionette-next

shrinktofit committed:
Animation graph: pose graph i18n related change (#15117)

shrinktofit committed:
Pose graph copy paste operation (#15132)

shrinktofit committed:
Event binding (#15133)

shrinktofit committed:
Marionette: opt state reenter opportunity to avoid essential problem (#15106)

shrinktofit committed:
Fix `poseGraphOp.disconnectNode` unexpected behavior (#15134)

bluesky013 commented on native/cocos/renderer/pipeline/custom/NativeResourceGraph.cpp:
```diff
@@ -159,6 +159,15 @@ gfx::TextureInfo getTextureInfo(const ResourceDesc& desc, bool bCube = false) {
     };
 }
 
+gfx::TextureViewInfo getTextureViewInfo(gfx::Texture* texture, const ResourceDesc& desc, uint32_t planeID) {
```
baseLayer 可能还不能跟 planeID 通用，应该要新增一个参数

shrinktofit committed:
Handle space conversion during pose graph evaluation, introduce base pose node class to modify pose (#15108)

* Process space conversion during pose graph evaluation

* Add comment

shrinktofit committed:
Animation graph: pose graph stash (#15050)

* Implement pose graph stash

* Update cocos/animation/marionette/pose-graph/stash/runtime-stash.ts

* Update cocos/animation/marionette/pose-graph/stash/runtime-stash.ts

* Rename

* Update

shrinktofit committed:
Pose graph: add blending pose nodes (#15142)

* Add blending nodes and choosing nodes

* i18n

shrinktofit committed:
Pose graph: add state machine node (#15145)

* Pose graph: add state machine node

* Import

* Add i18n

shrinktofit committed:
Pose graph: add pose modifying nodes (#15144)

* Add pose modifying nodes

* Update

* i18n

shrinktofit committed:
Add editor api to stash pose graph; Config some nodes' editor behavior (#15148)

shrinktofit committed:
Pose graph: add variable getting nodes (#15151)

* Pose graph: add variable getting nodes

* Fix unit test

shrinktofit committed:
Pose graph: add motion playing and sampling node (#15143)

* Add PlayMotionNode, SampleMotionNode

* Fix

shrinktofit committed:
Rename term "pose state" -> "procedural pose state" (#15171)

shrinktofit committed:
Pose graph: fix import path and tweak (#15173)

yiwenxue committed:
native external updates tag to develop-23 (#15172)

zvars committed:
fix tiled map object node layer & tileset from atlas supported (#13609)

* tiled map tileset import from atlas is supported.
fix node layer error created in TiledObjectGroup.

* CRLF -> LF

* revert tiled-object-group.ts imports changes

* revert tiled-object-group.ts imports changes

* fix tiled layer don't repaint when the camera has changed (position, size, etc.)

* this commit has move to branch 'tiled-layer-repait-fix'

* Better variable naming to follow convention

---------

Co-authored-by: pandamicro <lphuabin@gmail.com>

shrinktofit committed:
Add engine feature "procedural animation" (#15174)

AFeiYA committed:
Import gltf use vertex color set default to false (#15155)

doommm committed:
update: add bool type property to control `receiveShadow` (#15161)

- For UI consistent propose, add a new boolean property to control `receiveShadow`, and this property will be rendered as a `checkbox` in inspector

LorenzoDezi committed:
Windows devices  input fix (#14073)

* Windows devices  input fix

Pointer events are used

* Pull request fixes

Callback signatures and unnecessary condition statement

---------

Co-authored-by: Lorenzo Dezi <dezio95.ld@gmail.com>
Co-authored-by: pandamicro <lphuabin@gmail.com>

doommm committed:
fix: node, asset title text overflow in inspector (#15176)

- show ellipsis when title text has exceeded its parent container

shrinktofit committed:
[Bug] Pose graph: i18n menu (#15180)

shrinktofit committed:
renameObjectProperty() shall not copy non-enumerable symbol properties (#14194)

xubing0906 committed:
fix environment reflection not update when moving reflection probe (#15181)

dogodo-cc committed:
feat: inspector: fbx/animator  add import_all_animator config (#15159)

* feat: inspector: fbx/animator  add import_all_animator config

* feat: rename

* Update editor/i18n/en/assets.js

Co-authored-by: Leslie Leigh (李的序) <leslieleigh@outlook.com>

* Update editor/i18n/zh/assets.js

Co-authored-by: Leslie Leigh (李的序) <leslieleigh@outlook.com>

* feat: model change to animation

---------

Co-authored-by: Leslie Leigh (李的序) <leslieleigh@outlook.com>

moshuying committed:
Update code to adapt to memory leaks in the editor scene as a prerequisite task (#14594)

* revert code

* update code

arsen2010 committed:
optimize inspector interface  (#15177)

* format node.js

* fix inspector getName utils bug

* fix inspector node.js interface bug

* fix inspector particle-3d interface bug

* fix inspector edit asset interface bug

* add material fault tolerance

SantyWang committed:
remove repliated and client load (#13521)

VisualSJ committed:
Merge pull request #15175 from cocos/marionette-next

Pose Graph

star-e committed:
add ImageView

star-e committed:
Merge branch 'develop' into develop-pipeline

star-e committed:
add ImageView

LinYunMo committed:
Text Processing Refactoring (#14739)

* text processing file add

* processing String

* render & effect

* bmfont add

* char mode add

* code collate

* use less remove

* after rebase and change Sync

* add data class

* fix bm font clamp bug

* fix bitmap mode

* fix ci

* fix misc

* fix ci

* remove some code

* remove data class

* name change

* bug fix

star-e committed:
fix lint

star-e committed:
add textureView storage

star-e committed:
fix lint

troublemaker52025 committed:
fix hbao textures and rt format (#15184)

knoxHuang committed:
Fix wrong color type definition of blendColor in BlendStateEditor. (#15183)

* Fix wrong color type definition of blendColor in BlendStateEditor

* refine code

jk20012001 committed:
disable custom pipeline shadow map pass with native engine (#15186)

yiwenxue committed:
physics native removes seting fixedtimestep temporarily (#15187)

PPpro committed:
fix: undefined getWritableComponents on native platformed (#15190)

star-e committed:
adding temporary TestPipeline

knoxHuang committed:
Fix the type definition warning problem of blendColor (#15194)

star-e committed:
adding shadow

star-e committed:
Merge branch 'develop' into develop-pipeline

shrinktofit committed:
Animation graph: fix embedded play evaluation (#15198)

PPpro committed:
fix: use onWidnowResize on RUNTIME_BASED (#15201)

bofeng-song committed:
adapt xmlhttprequest  for taobao (#15199)

2youyou2 committed:
fixed depth slot name with custom pipeline (#15204)

jk20012001 committed:
fix post processing shader flip Y (#15205)

novain1102 committed:
fix wechatgame short black screen after first screen (#15209)

* fix wechatgame short black screen after first screen

* fix wechatgame firstScreen display misalignment

* add comment for ratio value

GengineJS committed:
Complete the separation of resource and pass setup and fix the problem of repeated push of resourceUses (#134)

star-e committed:
Merge branch 'v3.8.0' into develop-pipeline

star-e committed:
fix case

bluesky013 committed:
fix vulkan && metal subpass read-write. (#133)

hana-alice committed:
sync ds view
